test/study-webidl.js

```diff
@@ -361,12 +391,318 @@ interface mixin MyNamespaceMixin {};
+    assertEqual(report, 5, 'name', 'overloaded');
+    assertEqual(report, 5, 'message', '"operation overload" in partial interface mixin "MyRoom" overloads an operation defined in interface mixin "MyRoom"');
+  });
+
+
+  it('reports overloads across definitions (partial and mixin, different specs)', async () => {
+    const crawlResult = toCrawlResult(`
+[Global=Home,Exposed=*]
+interface MyHome {
+  undefined overload();
+};
+MyHome includes MyRoom;
+
+partial interface MyHome {
+  undefined overload(DOMString thing);
+};
+`).concat(toCrawlResult(`
+interface mixin MyRoom {
+  undefined overload(DOMString thing);
+};
+
+partial interface mixin MyRoom {
+  undefined overload(DOMString thing, boolean asap);
+};
+`, specUrl2));
+    const report = await studyWebIdl(crawlResult);
+
+    assertEqual(report.length, 6);
+    assertEqual(report, 0, 'name', 'overloaded');
+    assertEqual(report, 0, 'message', '"operation overload" in partial interface "MyHome" overloads an operation defined in interface "MyHome"');
+    assertEqual(report, 0, 'specs.length', 1);
+    assertEqual(report, 0, 'specs[0].url', specUrl);
+
+    assertEqual(report, 1, 'name', 'overloaded');
+    assertEqual(report, 1, 'message', `"operation overload" in interface "MyHome" overloads an operation defined in interface mixin "MyRoom" (in ${specUrl2})`);
+    assertEqual(report, 1, 'specs.length', 1);
+    assertEqual(report, 1, 'specs[0].url', specUrl);
+
+    assertEqual(report, 2, 'name', 'overloaded');
+    assertEqual(report, 2, 'message', `"operation overload" in partial interface "MyHome" overloads an operation defined in interface mixin "MyRoom" (in ${specUrl2})`);
+    assertEqual(report, 2, 'specs.length', 1);
+    assertEqual(report, 2, 'specs[0].url', specUrl);
+
+    assertEqual(report, 3, 'name', 'overloaded');
+    assertEqual(report, 3, 'message', `"operation overload" in interface "MyHome" overloads an operation defined in partial interface mixin "MyRoom" (in ${specUrl2})`);
+    assertEqual(report, 3, 'specs.length', 1);
+    assertEqual(report, 3, 'specs[0].url', specUrl);
+
+    assertEqual(report, 4, 'name', 'overloaded');
+    assertEqual(report, 4, 'message', `"operation overload" in partial interface "MyHome" overloads an operation defined in partial interface mixin "MyRoom" (in ${specUrl2})`);
+    assertEqual(report, 4, 'specs.length', 1);
+    assertEqual(report, 4, 'specs[0].url', specUrl);
+
+    assertEqual(report, 5, 'name', 'overloaded');
+    assertEqual(report, 5, 'message', '"operation overload" in partial interface mixin "MyRoom" overloads an operation defined in interface mixin "MyRoom"');
+    assertEqual(report, 5, 'specs.length', 1);
+    assertEqual(report, 5, 'specs[0].url', specUrl2);
+  });
+
+
+  it('reports redefined members (same spec)', async () => {
+    const report = await analyzeIdl(`
+[Global=Home,Exposed=*]
+interface MyHome {
+  undefined overload();
```

Comment:
nit: maybe use a different name than `overload` (e.g. `redefined`)

Reply:
If I can't even create a bad WWII pun with a (misnamed) "operation overload"... :(
(Fun fact: [operation overload](https://en.wikipedia.org/wiki/Operation_Overload) seems to exist in practice, I didn't know ;))

Reply:
you can keep the pun for the overload test; you just have to find a different pun for redefined :)

Reply:
Ah, I hadn't spotted the difference. Good, then `dejaVu` it is :)